packages/neon-core/src/sc/ContractParam.ts

```diff
@@ -81,144 +92,235 @@ export class ContractParam implements NeonObject<ContractParamLike> {
-        case ContractParamType.PublicKey:
-          this.value = HexString.fromHex(input.value as string);
+        } else {
+          throw new Error("Please provide a HexString for value!");
+        }
+
+      case ContractParamType.Integer:
+      case ContractParamType.String:
+        if (typeof arg === "string") {
+          this.value = arg;
           return;
-        case ContractParamType.Integer:
-        case ContractParamType.String:
-          this.value = input.value as number | string;
+        } else {
+          throw new Error("Please provide a string for value!");
+        }
+
+      case ContractParamType.Array:
+        if (Array.isArray(arg)) {
+          this.value = (arg as (
+            | ContractParam
+            | ContractParamLike
+          )[]).map((i: ContractParam | ContractParamLike) =>
+            ContractParam.fromJson(i)
+          );
           return;
-        case ContractParamType.Any:
-          this.value =
-            typeof input.value === "object"
-              ? (input.value as (ContractParamLike | ContractParam)[]).map(
-                  (cp) => new ContractParam(cp)
-                )
-              : input.value ?? null;
+        } else {
+          throw new Error("Please provide an array for value!");
+        }
+
+      case ContractParamType.Void:
+        if (arg === null || arg === undefined) {
+          this.value = null;
           return;
-      }
-      this.value =
-        typeof input.value === "object"
-          ? (input.value as (ContractParamLike | ContractParam)[]).map(
-              (cp) => new ContractParam(cp)
-            )
-          : input.value ?? null;
-    } else {
-      throw new Error("No constructor arguments provided!");
+        } else {
+          throw new Error("Void should not have any value provided.");
+        }
+
+      default:
+        throw new Error(`${this.type} not supported!`);
     }
   }
 
   public get [Symbol.toStringTag](): string {
     return "ContractParam:" + ContractParamType[this.type];
   }
 
-  public export(): ContractParamLike {
+  /**
+   * Creates a ContractParam from a JSON representation. Use this as the main entry point for conversion from external systems.
+   * @param json - JSON format
+   */
+  public static fromJson(json: ContractParamLike): ContractParam {
+    if (json instanceof ContractParam) {
+      return new ContractParam(json);
+    }
+    const type = parseEnum(json.type, ContractParamType);
+    const arg = json.value;
+    switch (type) {
+      case ContractParamType.Array:
+        if (Array.isArray(arg)) {
+          return ContractParam.array(...arg);
+        }
+        break;
+      case ContractParamType.Boolean:
+        if (
+          typeof arg === "string" ||
+          typeof arg === "number" ||
+          typeof arg === "boolean"
+        ) {
+          return ContractParam.boolean(arg);
+        }
+        break;
+
+      case ContractParamType.ByteArray:

```

Comment:
bytearray (and `signature`) are now base64 encoded
https://github.com/neo-project/neo/blob/7d3a4a2667a2c35db0b775c02082ec03e05cfa65/src/neo/SmartContract/ContractParameter.cs#L148-L151

Reply:
The tests are all on preview3. I will upgrade the integration test container when the heads are buildable (sometimes they are not) but for now, Im sticking to preview3 compatible code which is testable.

Reply:
Are you sure you tested this against preview3? Because this change (https://github.com/neo-project/neo/pull/1199) was introduced 12 months ago, whereas preview3 was released 3 months ago.

Reply:
updated to use base64.

Reply:
👍